cmd/agent/subcommands/run/internal/clcrunnerapi/v1/clcrunner.go

```diff
@@ -70,6 +72,29 @@ func flattenCLCStats(stats status.CLCChecks) map[string]status.CLCStats {
 	return flatened
 }
 
+// replaceIDsWithIDsKnownByDCA replaces the check IDs in the map received with
+// the ID that those checks had before decrypting their secrets. This is needed
+// because if the Cluster Agent does not decrypt secrets and the runner does,
+// the check ID seen by both of them is going to be different and the Cluster
+// Agent won't recognize the check as a cluster check.
+// The API defined in this file is only used by the Cluster Agent, so it makes
+// sense to use the IDs that it recognizes.
+func replaceIDsWithIDsKnownByDCA(stats map[string]status.CLCStats) map[string]status.CLCStats {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Any reason to return a copy instead of doing the change in-place ?
I guess you could even do it directly in `flattenCLCStats` if it's not used elsewhere

Reply:
I did it just for clarity. It's true that it's not optimal but I don't think it can have an impact in this case because this endpoint is called every 10 min.